File: BACK-END/src/main/java/org/kuenteco/backend/dto/logic/budget/BudgetSummaryDTO.java
package org.kuenteco.backend.dto.logic.budget;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class BudgetSummaryDTO {
    private String userId;
    private String ownerUserId; // Para consistencia con las vistas
    private String username;
    private Long totalBudgets;
    private BigDecimal totalBudgetAmount;
    private BigDecimal totalRemainingAmount;
    private BigDecimal totalSpentAmount;
    private BigDecimal overallPercentageUsed;
}

File: FRONT-END/lib/dto/budget/budget_summary_dto.dart
import 'package:decimal/decimal.dart';

class BudgetSummaryDTO {
  final String userId;
  final String ownerUserId;
  final String username;
  final int totalBudgets;
  final Decimal totalBudgetAmount;
  final Decimal totalRemainingAmount;
  final Decimal totalSpentAmount;
  final Decimal overallPercentageUsed;

  BudgetSummaryDTO({
    required this.userId,
    required this.ownerUserId,
    required this.username,
    required this.totalBudgets,
    required this.totalBudgetAmount,
    required this.totalRemainingAmount,
    required this.totalSpentAmount,
    required this.overallPercentageUsed,
  });

  factory BudgetSummaryDTO.fromJson(Map<String, dynamic> json) {
    return BudgetSummaryDTO(
      userId: json['userId'],
      ownerUserId: json['ownerUserId'],
      username: json['username'],
      totalBudgets: json['totalBudgets'],
      totalBudgetAmount: Decimal.parse(json['totalBudgetAmount'].toString()),
      totalRemainingAmount: Decimal.parse(json['totalRemainingAmount'].toString()),
      totalSpentAmount: Decimal.parse(json['totalSpentAmount'].toString()),
      overallPercentageUsed: Decimal.parse(json['overallPercentageUsed'].toString()),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'userId': userId,
      'ownerUserId': ownerUserId,
      'username': username,
      'totalBudgets': totalBudgets,
      'totalBudgetAmount': totalBudgetAmount.toString(),
      'totalRemainingAmount': totalRemainingAmount.toString(),
      'totalSpentAmount': totalSpentAmount.toString(),
      'overallPercentageUsed': overallPercentageUsed.toString(),
    };
  }
}
